client/src/analytics_component.rs

```diff
@@ -1,68 +1,520 @@
+
+                let style = format!(
+                    "left: {}%; top: {}%; background: {}; opacity: {};",
+                    x,
+                    y,
+                    background,
+                    // Fade dots near the edges
+                    if x < 30.0 || x > 70.0 { 0.5 } else { 1.0 }
+                );
+
+                html! {
+                    <div class="visitor-dot" {style}>
+                        <div class="visitor-pulse"></div>
+                    </div>
+                }
+            } else {
+                html! {}
+            }
+        })
+        .collect::<Html>();
+
+    html! {
+        <div class="ascii-globe">
+            <div class="globe-container">
+                <pre ref={globe_ref} class="globe-text"></pre>
+                { visitor_dots }
+            </div>
+            <div class="globe-legend">
+                <div class="legend-item active">
+                    <span class="dot"></span>
+                    <span>{"Active Visitors"}</span>
+                </div>
+                <div class="legend-item recent">
+                    <span class="dot"></span>
+                    <span>{"Recent Visitors"}</span>
+                </div>
+            </div>
+        </div>
+    }
+}
+
 #[function_component(AnalyticsContent)]
 fn analytics_content() -> HtmlResult {
+    let view_type = use_state(|| ViewType::Table);
+    let country_filter = use_state(String::new);
+    let search_filter = use_state(String::new);
+    let sort_field = use_state(|| SortField::Visits);
+    let sort_direction = use_state(|| SortDirection::Descending);
+    let current_page = use_state(|| 1);
+    let items_per_page = 10;
+
     let resp = use_future(|| async {
         Request::get("/api/analytics")
             .send()
             .await?
             .json::<Vec<Analytics>>()
             .await
     })?;
+
     match *resp {
         Ok(ref res) => {
-            let analytics: Html = res.iter().map(to_html).collect();
+            let mut filtered_data = res
+                .iter()
+                .filter(|a| {
+                    let country_match =
+                        country_filter.is_empty() || &a.iso_code == &*country_filter;
+                    let search_match = search_filter.is_empty()
+                        || a.ip_address
+                            .to_lowercase()
+                            .contains(&search_filter.to_lowercase())
+                        || a.path
+                            .to_lowercase()
+                            .contains(&search_filter.to_lowercase());
+                    country_match && search_match
+                })
+                .cloned()
+                .collect::<Vec<Analytics>>();
+
+            // Sort the data
+            sort_analytics(&mut filtered_data, &sort_field, &sort_direction);
+
+            // Calculate pagination
+            let total_pages = (filtered_data.len() as f32 / items_per_page as f32).ceil() as usize;
```

Comment:
`(filtered_data.len() + (items_per_page - 1))/items_per_page` is probably better